bitrise/publish-internal-podspec.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+## I. Check parameter
+if [ -z $1 ] || ( [ "$1" != "internal" ] && [ "$1" != "external" ] ); then
+  echo "Invalid parameter.";
+  echo "  Usage: $0 {internal|external}";
+  exit 1;
+fi
+
+## II. Get publish version for information
+publish_version="$(grep "VERSION_STRING" $BITRISE_SOURCE_DIR/$VERSION_FILE | head -1 | awk -F "[= ]" '{print $4}')"
+echo "Publishing podspec for version" $publish_version
+
+if [ "$1" == "internal" ]; then
+
+  ## 1. Get path of internal podspec local repo
+  local_repo_path="$(pod repo | grep "$GIT_SPEC_REPO_NAME" | grep Path | head -1 | awk -F ": " '{print $2}')"
+
+  ## 2. Update podspec to the internal podspec local repo
+  resp="$(pod repo push $GIT_SPEC_REPO_NAME $BITRISE_SOURCE_DIR/$PODSPEC_FILENAME --allow-warnings)"
+  echo $resp
+
+  # Check error from the response
+  error="$(echo $resp | grep -i error\|fatal)"
+  if [ "$error" ]; then
+    echo "Cannot publish to internal repo"
+    exit 1
+  fi
+
+  ## 3. Push podspec to the internal podspec remote repo
+  cd $local_repo_path
+  git push
+  cd $BITRISE_SOURCE_DIR
+
+  echo "Podspec published to internal repo successfully"
+
+else
+
+  ## 1. Run lint to validate podspec. This step will automatically push the spec to CocoaPods if the spec is valid
+  resp="$(pod spec lint $BITRISE_SOURCE_DIR/$PODSPEC_FILENAME --allow-warnings)"
```

Comment:
`pod spec lint` does not automatically publish a spec to cocoapods. You have to run `trunk` for that. I'm validation podspecs with `pod spec lint` several times locally before I call `pod trunk push` or whatever the command for publishing is.